tempesta_fw/cache.c

```diff
@@ -1388,34 +1388,39 @@ tfw_cache_purge_invalidate(TfwHttpReq *req)
 /**
  * Process PURGE request method according to the configuration.
  */
-static int
+static void
 tfw_cache_purge_method(TfwHttpReq *req)
 {
-	int ret;
 	TfwAddr saddr;
 	TfwVhost *vhost = tfw_vhost_get_default();
 
 	/* Deny PURGE requests by default. */
-	if (!(cache_cfg.cache && vhost->cache_purge && vhost->cache_purge_acl))
-		return tfw_http_send_403(req, "purge: not configured");
+	if (!(cache_cfg.cache && vhost->cache_purge && vhost->cache_purge_acl)) {
+		tfw_http_send_403(req, "purge: not configured");
+		return;
+	}
 
 	/* Accept requests from configured hosts only. */
 	ss_getpeername(req->conn->sk, &saddr);
-	if (!tfw_capuacl_match(vhost, &saddr))
-		return tfw_http_send_403(req, "purge: ACL violation");
+	if (!tfw_capuacl_match(vhost, &saddr)) {
+		tfw_http_send_403(req, "purge: ACL violation");
+		return;
+	}
 
 	/* Only "invalidate" option is implemented at this time. */
 	switch (vhost->cache_purge_mode) {
 	case TFW_D_CACHE_PURGE_INVALIDATE:
-		ret = tfw_cache_purge_invalidate(req);
+		if (tfw_cache_purge_invalidate(req)){
+			tfw_http_send_404(req, "purge: processing error");
+			return;
+		}
 		break;
 	default:
-		return tfw_http_send_403(req, "purge: invalid option");
+		tfw_http_send_403(req, "purge: invalid option");
+		return;
 	}
 
-	return ret
-		? tfw_http_send_404(req, "purge: processing error")
-		: tfw_http_send_200(req);
+	tfw_http_send_200(req);
```

Comment:
I personally like the old approach in this switch-case block, when the error processing was divided from handling different purge methods.

Reply:
Corrected.